src/rapidhtml/components/table/__init__.py

```diff
@@ -0,0 +1,65 @@
+from typing import Callable, Mapping
+
+from rapidhtml import tags as html_tags
+from rapidhtml.callbacks import RapidHTMLCallback
+
+
+class Table(html_tags.BaseTag):
+    def __init__(self, callback: Callable | RapidHTMLCallback = None, **attrs) -> None:
+        super().__init__("table", callback=callback, **attrs)
+
+        self.__columns: list[str] = []
+        self.__data: list[Mapping[str, int | float | str]] = []
+
+    @property
+    def columns(self) -> list[str]:
+        return self.__columns
+
+    @property
+    def rows(self) -> list[Mapping[str, int | float | str]]:
+        return self.__data
+
+    @columns.setter
+    def columns(self, columns: list[str]) -> None:
+        if self.__data:
+            raise ValueError(
+                "Cannot set columns after data has been added. Call clear_rows() first."
+            )
+        self.__columns = columns
+
+    def add_row(self, row: list[str]) -> None:
+        padding = [""] * (len(self.__columns) - len(row))
+        padded_row = row + padding
+
+        current_row_data = {
+            column_name: row_data
+            for column_name, row_data in zip(self.__columns, padded_row)
+        }
+        self.__data.append(current_row_data)
+
+    def add_rows(self, *rows: list[str]) -> None:
```

Comment:
Definitely needs a usage example. My first instinct would be to pass in `list[list[str]]`. I like the way it is currently more than my instinct, just needs to be documented